### DISCUSSION TITLE
What does `${@:3}` mean in `dist_train.sh`?

### DISCUSSION THREAD

**Discussion by songyuc** [open-mmlab/mmdetection#8906] [Q&A] [answered]
Hi, guys!
I am learning MMDetection today, and I am curious about the meaning of `${@:3}` mean in `dist_train.sh`, as [[code]](https://github.com/open-mmlab/mmdetection/blob/9d3e162459590eee4cfc891218dfbb5878378842/tools/dist_train.sh#L20).

Your answer and guide will be appreciated!

**Comment by i-aki-y** [comment, marked as answer]
${@} is a special parameter that contains whole positional parameters ($1, $2, ...)
${@:n} means expand the array from n-th parameter to the end ($n, $n+1, ...)
For more detail, see the "${parameter:offset}" section in:
https://www.gnu.org/software/bash/manual/bash.html#Shell-Parameter-Expansion

**Reply by songyuc** [reply]
Thank you so much!
